### DISCUSSION TITLE
worker resources

### DISCUSSION THREAD

**Discussion by vqoley** [giongto35/cloud-game#411] [Q&A]
i like to host on server. can i know this source code, when play game its using which resources? ram? processor? gpu?

also how many resources that use for active and inactive worker?

**Comment by sergystepanov** [comment]
The app itself won't track and show utilization to users.
I'd recommend to install an external real-time monitoring service onto your server (for example, Netdata) or just run a process monitor like top, htop, btop.
Roughly: one coordinator instance and 5 workers will require 350MiB of RAM dockerized on Linux, and that may grow 30% during gaming due to Go runtime requirements + another 30%-200% **on each worker** depending on the currently running Libretro core (emulator). Mupen64 easily eats ~250MiB (or 900MiB on Windows).
For CPU, take a look at the bench results for cloudretro.io:
```
# ## ## ## ## ## ## ## ## ## ## ## ## ## ## ## ## ## #
#              Yet-Another-Bench-Script              #
#                     v2023-09-06                    #
# https://github.com/masonr/yet-another-bench-script #
# ## ## ## ## ## ## ## ## ## ## ## ## ## ## ## ## ## #

Tue Sep 26 14:58:59 UTC 2023

Basic System Information:
---------------------------------
Uptime     : 0 days, 0 hours, 25 minutes
Processor  : DO-Regular
CPU cores  : 1 @ 2199.996 MHz
AES-NI     : ✔ Enabled
VM-x/AMD-V : ✔ Enabled
RAM        : 957.5 MiB
Swap       : 2.0 GiB
Disk       : 24.1 GiB
Distro     : Ubuntu 22.04.3 LTS
Kernel     : 5.15.0-84-generic
VM Type    : KVM
IPv4/IPv6  : ✔ Online / ❌ Offline

IPv4 Network Information:
---------------------------------
ISP        : DigitalOcean, LLC
ASN        : AS14061 DigitalOcean, LLC
Host       : DigitalOcean, LLC
Location   : Santa Clara, California (CA)
Country    : United States

Running GB6 benchmark test... *cue elevator music*
Geekbench 6 Benchmark Test:
---------------------------------
Test            | Value
                |
Single Core     | 467
Multi Core      | 361
Full Test       | https://browser.geekbench.com/v6/cpu/2783259

YABS completed in 24 min 17 sec
```
This CPU can run concurrently only NES/SNES/GBA emulators and only one PSX/N64 game. You can approximate your hardware accordingly.
And normally GPU is not used in cloud-game, unless you run workers outside the Docker containers, but I didn't use it with a GPU apart from basic tests, so can't tell much here.
